src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -593,16 +597,23 @@ internal void ThrowIfNotOpenable(bool needToUncompress, bool needToLoadIntoMemor
         private CheckSumAndSizeWriteStream GetDataCompressor(Stream backingStream, bool leaveBackingStreamOpen, EventHandler onClose)
         {
             // stream stack: backingStream -> DeflateStream -> CheckSumWriteStream
-
-            // we should always be compressing with deflate. Stored is used for empty files, but we don't actually
-            // call through this function for that - we just write the stored value in the header
-            Debug.Assert(CompressionMethod == CompressionMethodValues.Deflate);
-
-            Stream compressorStream = _compressionLevel.HasValue ?
-                new DeflateStream(backingStream, _compressionLevel.Value, leaveBackingStreamOpen) :
-                new DeflateStream(backingStream, CompressionMode.Compress, leaveBackingStreamOpen);
+            Debug.Assert(CompressionMethod == CompressionMethodValues.Deflate || CompressionMethod == CompressionMethodValues.Stored);
 
             bool isIntermediateStream = true;
+            Stream compressorStream;
```

Comment:
```text
// we should always be compressing with deflate. Stored is used for empty files, but we don't actually	
// call through this function for that - we just write the stored value in the header
```


Can we keep and update the relevant pieces of the above comment that got removed?

**Something like:**
```text
If `CompressionMethod` is set to something other than `Stored`, we should always be compressing with deflate.
However, if `_compressionLevel` is `NoCompression`, do not use `DeflateStream`.
```